src/DependencyInjection/DI/src/ServiceProvider.cs

```diff
@@ -27,6 +28,21 @@ internal ServiceProvider(IEnumerable<ServiceDescriptor> serviceDescriptors, Serv
 
             switch (options.Mode)
             {
+                case ServiceProviderMode.Default:
+#if !NETCOREAPP
+                    _engine = new DynamicServiceProviderEngine(serviceDescriptors, callback);
+#else
+                    if (RuntimeFeature.IsSupported("IsDynamicCodeCompiled"))
```

Comment:
Isn't this a constant?

Reply:
Not in `netcoreapp2.0`, I can change to cross-compile to `netcoreapp3.0` but we'll lose IL_EMIT on downlevel netcoreapps